File: app/src/main/java/com/iodraw/beijingclock/MainActivity.java
package com.iodraw.beijingclock;

import android.app.Activity;
import android.graphics.Color;
import android.os.Bundle;
import android.os.Handler;
import android.os.Looper;
import android.util.TypedValue;
import android.view.Gravity;
import android.widget.TextView;

import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Locale;
import java.util.TimeZone;

public class MainActivity extends Activity {
    private final Handler handler = new Handler(Looper.getMainLooper());
    private final SimpleDateFormat timeFormat = new SimpleDateFormat("HH:mm:ss", Locale.CHINA);
    private TextView clockView;

    private final Runnable tick = new Runnable() {
        @Override
        public void run() {
            clockView.setText(timeFormat.format(new Date()));
            handler.postDelayed(this, 1000);
        }
    };

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        timeFormat.setTimeZone(TimeZone.getTimeZone("Asia/Shanghai"));

        clockView = new TextView(this);
        clockView.setBackgroundColor(Color.parseColor("#111111"));
        clockView.setTextColor(Color.parseColor("#F5F5F5"));
        clockView.setGravity(Gravity.CENTER);
        clockView.setTextSize(TypedValue.COMPLEX_UNIT_SP, 56);
        clockView.setLetterSpacing(0.08f);

        setContentView(clockView);
    }

    @Override
    protected void onStart() {
        super.onStart();
        tick.run();
    }

    @Override
    protected void onStop() {
        super.onStop();
        handler.removeCallbacks(tick);
    }
}

File: app/src/main/java/com/iodraw/beijingclock/MainActivity.kt
package com.iodraw.beijingclock

import android.os.Bundle
import android.os.Handler
import android.os.Looper
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale
import java.util.TimeZone

class MainActivity : AppCompatActivity() {
    private val handler = Handler(Looper.getMainLooper())
    private lateinit var timeTextView: TextView

    private val formatter = SimpleDateFormat("HH:mm:ss", Locale.CHINA).apply {
        timeZone = TimeZone.getTimeZone("Asia/Shanghai")
    }

    private val ticker = object : Runnable {
        override fun run() {
            timeTextView.text = formatter.format(Date())
            handler.postDelayed(this, 1000)
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        timeTextView = findViewById(R.id.timeText)
    }

    override fun onStart() {
        super.onStart()
        ticker.run()
    }

    override fun onStop() {
        super.onStop()
        handler.removeCallbacks(ticker)
    }
}
